workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityComponentDataGenerator.tt

```diff
@@ -152,5 +156,37 @@ namespace <#= qualifiedNamespace #>
                 get => <#= qualifiedNamespace #>.<#= componentDetails.ComponentName #>.ReferenceTypeProviders.UpdatesProvider.Get(handle);
             }
         }
+
+        internal class <#= componentDetails.ComponentName  #>Dynamic : IDynamicInvokable
+        {
+            public uint ComponentId => <#= componentDetails.ComponentName  #>.ComponentId;
+
+            private static Component DeserializeData(ComponentData data, World world)
+            {
+                var schemaDataOpt = data.SchemaData;
+                if (!schemaDataOpt.HasValue)
+                {
+                    throw new ArgumentException($"Can not deserialize an empty {nameof(ComponentData)}");
+                }
```

Comment:
I don't _think_ these checks are required. I think the only reason its nullable is that if you pass in an actual vtable you will get the result of that instead of the raw schema data

Reply:
Discussed offline - protection against users passing in their own potentially malformed ops and reduces boilerplate when using the deserialize functions

Reply:
Yeah, this is really a marginal check. Arguably this is no better than the error you'd get if this just fell through though.